homeassistant/components/vesync/light.py

```diff
@@ -51,34 +57,149 @@ def _async_setup_entities(devices, async_add_entities):
+                result,
+            )
+            return 0
+        # convert percent brightness to ha expected range
+        return round((max(1, brightness_value) / 100) * 255)
 
     def turn_on(self, **kwargs):
         """Turn the device on."""
-        if ATTR_BRIGHTNESS in kwargs:
+        attribute_adjustment_only = False
+        # set white temperature
+        if self.color_mode in (COLOR_MODE_COLOR_TEMP) and ATTR_COLOR_TEMP in kwargs:
+            # get white temperature from HA data
+            color_temp = int(kwargs[ATTR_COLOR_TEMP])
+            # ensure value between min-max supported Mireds
+            color_temp = max(self.min_mireds, min(color_temp, self.max_mireds))
+            # convert Mireds to Percent value that api expects
+            color_temp = round(
+                ((color_temp - self.min_mireds) / (self.max_mireds - self.min_mireds))
+                * 100
+            )
+            # flip cold/warm to what pyvesync api expects
+            color_temp = 100 - color_temp
+            # ensure value between 0-100
+            color_temp = max(0, min(color_temp, 100))
+            # call pyvesync library api method to set color_temp
+            self.device.set_color_temp(color_temp)
+            # flag attribute_adjustment_only, so it doesn't turn_on the device redundantly
+            attribute_adjustment_only = True
+        # set brightness level
+        if (
+            self.color_mode in (COLOR_MODE_BRIGHTNESS, COLOR_MODE_COLOR_TEMP)
+            and ATTR_BRIGHTNESS in kwargs
+        ):
             # get brightness from HA data
             brightness = int(kwargs[ATTR_BRIGHTNESS])
+            # ensure value between 1-255
+            brightness = max(1, min(brightness, 255))
             # convert to percent that vesync api expects
             brightness = round((brightness / 255) * 100)
-            # clamp to 1-100
+            # ensure value between 1-100
             brightness = max(1, min(brightness, 100))
-            self.dimmer.set_brightness(brightness)
-        # Avoid turning device back on if this is just a brightness adjustment
-        if not self.is_on:
-            self.device.turn_on()
+            # call pyvesync library api method to set brightness
+            self.device.set_brightness(brightness)
+            # flag attribute_adjustment_only, so it doesn't turn_on the device redundantly
+            attribute_adjustment_only = True
+        # check flag if should skip sending the turn_on command
+        if attribute_adjustment_only:
+            return
+        # send turn_on command to pyvesync api
+        self.device.turn_on()
+
+
+class VeSyncDimmableLightHA(VeSyncBaseLight, LightEntity):
+    """Representation of a VeSync dimmable light device."""
+
+    def __init__(self, device):
+        """Initialize the VeSync dimmable light device."""
+        super().__init__(device)
+        self.device = device
```

Comment:
Since the parent class sets this, you can drop setting it here


Actually probably the whole Init block since pylint Will complains it's a useless delegation

Reply:
that __init__ part is something I'm learning as a go. It was there before, thne,  since you asked to remove it, I did, and tested, things start to make more sense to me now. thanks